web_sdk/BUILD.gn

```diff
@@ -77,6 +78,11 @@ template("sdk_rewriter") {
       "--stamp",
       rebase_path(stamp_location, root_build_dir),
     ]
+
+    if (defined(invoker.is_public) && invoker.is_public) {
+      args += [ "--public" ]
+    }
```

Comment:
It looks like this flag only affects the library name `$libraryName` vs `_$libraryName`. Could we just change library names (e.g. `library_name = "_engine"`) instead of introducing a new flag?

Reply:
Not without doing a ton of changes across the codebase that actually change the original source files to all be underscore prefixed (and I think there is probably a reason it is this way, having to do with how package imports work vs library imports). The difference is that the original source files all treat it as `engine` and the transformed files treat it as `_engine`. It's not just a matter of changing the library name, because there is actually an "input library name" and "output library name". In the private case, that means adding an underscore, in the public case they are the same. So I can either have a `public` flag that tells us whether to add the underscore or not, or I can have an `inputLibraryName` and `outputLibraryName` or something like that that is specified by each transformation, but that seems too weird and generic and easy to mess up.